Tests/Realm.Tests/Sync/UserManagementTests.cs

```diff
@@ -1166,11 +1166,11 @@ public void UserCustomDataChange_RaisesChanged()
 
                 await tcs.Task;
 
-                tcs = new();
-
                 // Unsubscribe and verify that it no longer raises user changed
                 user.Changed -= OnUserChanged;
 
+                tcs = new();
```

Comment:
Sometimes no `TimeoutException` was raised at the end of the method on CI. I verified that actually we get two `Changed` notifications, one for the custom data, and one for the user token refresh. 
I suspect that sometimes the second notification was being raised after the new token completion source was created but before the unsubscription, that's why the timeout had no effect.